File: back_end/video-service/src/main/java/org/example/videoservice/service/VideoService.java
package org.example.videoservice.service;

import org.example.videoservice.dto.VideoDTO;
import org.example.videoservice.model.Category;
import org.example.videoservice.model.Type;

import java.util.List;

public interface VideoService {
    VideoDTO createVideo(VideoDTO videoDTO);

    VideoDTO updateVideo(Long id, VideoDTO videoDTO);

    VideoDTO getVideoById(Long id);

    List<VideoDTO> getAllVideos();

    List<VideoDTO> getVideosByType(Type type);

    List<VideoDTO> getVideosByCategory(Category category);

    void deleteVideo(Long id);
}

File: front_end/src/services/videoService.js
import api from './api';

export const videoService = {
    getAllVideos: async () => {
        const response = await api.get('/videos');
        return response.data;
    },

    getVideoById: async (id) => {
        const response = await api.get(`/videos/${id}`);
        return response.data;
    },

    getVideosByType: async (type) => {
        const response = await api.get(`/videos/type/${type}`);
        return response.data;
    },

    getVideosByCategory: async (category) => {
        const response = await api.get(`/videos/category/${category}`);
        return response.data;
    },

    createVideo: async (videoData) => {
        const response = await api.post('/videos', videoData);
        return response.data;
    },

    updateVideo: async (id, videoData) => {
        const response = await api.put(`/videos/${id}`, videoData);
        return response.data;
    },

    deleteVideo: async (id) => {
        const response = await api.delete(`/videos/${id}`);
        return response.data;
    }
};
